### PR TITLE
[AGENT-4700] Add support for DataRobot Credentials update from Airflow Connection Tab

### PR DESCRIPTION
**Pull request by oleksandr-saienko**
# This repository is public. Do not put here any private DataRobot or customer's data: code, datasets, model artifacts, .etc.

## Summary
Extended Airflow DataRobot hooks to support updating credentials after creation,
currently its updating credentials each time when GetOrCreateCredentials method invoked

## Rationale
Users should be able to update DataRobot Credentials/ Connections from Airflow Connection Tab

### CODE REVIEW

**Comment by oleksandr-saienko** [comment]
Jenkins run

**Comment by oleksandr-saienko** [comment]
Jenkins run

**Comment by oleksandr-saienko** [comment]
Jenkins run

**Comment by oleksandr-saienko** [comment]
Jenkins run

**Comment by oleksandr-saienko** [comment]
Jenkins run

File: datarobot_provider/hooks/credentials.py [hunk 1]
```diff
@@ -370,6 +453,22 @@ def create_credentials(self, conn) -> Credential:
                 f"Error creating Azure Credentials: {self.datarobot_credentials_conn_id}"
             )
 
+    def update_credentials(self, conn, credential: Credential) -> None:
+        """Updates Azure Storage credentials for params in connection object"""
+
+        azure_credentials = {"azure_connection_string": self.create_azure_connection_string(conn)}
+        self.log.info(f"Updating Azure Storage Credentials:{self.datarobot_credentials_conn_id}")
```
Comment (dulcardo) [inline datarobot_provider/hooks/credentials.py:460]:
```suggestion
        self.log.info(f"Updating Azure Storage Credentials: {self.datarobot_credentials_conn_id}")
```

File: datarobot_provider/hooks/credentials.py [hunk 2]
```diff
@@ -445,6 +544,33 @@ def create_credentials(self, conn) -> Credential:
...
+        refresh_token = conn.extra_dejson.get('refresh_token', '')
+        if not refresh_token:
+            raise AirflowException("OAuth Refresh Token is not defined")
+
+        oauth_credentials = {
+            "token": conn.password,
+            "refresh_token": refresh_token,
+        }
+
+        self.log.info(f"Updating OAuth Credentials:{self.datarobot_credentials_conn_id}")
```
Comment (dulcardo) [inline datarobot_provider/hooks/credentials.py:562]:
```suggestion
        self.log.info(f"Updating OAuth Credentials: {self.datarobot_credentials_conn_id}")
```

**Review by dulcardo — APPROVED** [review]

**Pull request merged**